src/NHibernate.Test/NHSpecificTest/DtcFailures/DtcFailuresFixture.cs

```diff
@@ -17,247 +17,308 @@ namespace NHibernate.Test.NHSpecificTest.DtcFailures
-		{
-			get { return new[] {"NHSpecificTest.DtcFailures.Mappings.hbm.xml"}; }
-		}
+			=> new[] { "NHSpecificTest.DtcFailures.Mappings.hbm.xml" };
 
 		protected override string MappingsAssembly
-		{
-			get { return "NHibernate.Test"; }
-		}
+			=> "NHibernate.Test";
 
 		protected override bool AppliesTo(Dialect.Dialect dialect)
-		{
-			return TestDialect.GetTestDialect(dialect).SupportsDistributedTransactions;
-		}
+			=> dialect.SupportsDistributedTransactions;
 
-        protected override void CreateSchema()
-        {
-            // Copied from Configure method.
-            Configuration config = new Configuration();
+		protected override void CreateSchema()
+		{
+			// Copied from Configure method.
+			var config = new Configuration();
 			if (TestConfigurationHelper.hibernateConfigFile != null)
 				config.Configure(TestConfigurationHelper.hibernateConfigFile);
 
-            // Our override so we can set nullability on database column without NHibernate knowing about it.
-            config.BeforeBindMapping += BeforeBindMapping;
+			// Our override so we can set nullability on database column without NHibernate knowing about it.
+			config.BeforeBindMapping += BeforeBindMapping;
 
-            // Copied from AddMappings methods.
-			Assembly assembly = Assembly.Load(MappingsAssembly);
-			foreach (string file in Mappings)
+			// Copied from AddMappings methods.
+			var assembly = Assembly.Load(MappingsAssembly);
+			foreach (var file in Mappings)
 				config.AddResource(MappingsAssembly + "." + file, assembly);
 
-            // Copied from CreateSchema method, but we use our own config.
-            new SchemaExport(config).Create(false, true);
-        }
-
-        private void BeforeBindMapping(object sender, BindMappingEventArgs e)
-        {
-            HbmProperty prop = e.Mapping.RootClasses[0].Properties.OfType<HbmProperty>().Single(p => p.Name == "NotNullData");
-            prop.notnull = true;
-            prop.notnullSpecified = true;
-        }
+			// Copied from CreateSchema method, but we use our own config.
+			new SchemaExport(config).Create(false, true);
+		}
 
-		[Test]
-		public void WillNotCrashOnDtcPrepareFailure()
+		protected override void OnTearDown()
 		{
-			var tx = new TransactionScope();
-			using (ISession s = sessions.OpenSession())
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
 			{
-				s.Save(new Person {NotNullData = null});  // Cause a SQL not null constraint violation.
+				s.CreateQuery("delete from System.Object").ExecuteUpdate();
+				t.Commit();
 			}
+		}
 
-			new ForceEscalationToDistributedTx();
+		private void BeforeBindMapping(object sender, BindMappingEventArgs e)
+		{
+			var prop = e.Mapping.RootClasses[0].Properties.OfType<HbmProperty>().Single(p => p.Name == "NotNullData");
+			prop.notnull = true;
+			prop.notnullSpecified = true;
+		}
 
-			tx.Complete();
-			try
+		[Test]
+		public void SupportsEnlistingInDistributed()
```

Comment:
Test only for that. Most DTC tests are not failing in a very explicit manner when DTC is not supported, because of most of them acquire the connection only during prepare phase, not before, causing the distributed failure to occur quite late.